core/LSU.hpp

```diff
@@ -319,19 +339,55 @@ namespace olympia
         void handleCacheReadyReq_(const MemoryAccessInfoPtr &memory_access_info_ptr);
         void getAckFromCache_(const MemoryAccessInfoPtr &updated_memory_access_info_ptr);
 
+        // Perform cache read
+        void handleCacheRead_();
         // Retire load/store instruction
         void completeInst_();
 
         // Handle instruction flush in LSU
         void handleFlush_(const FlushCriteria &);
 
+        // Instructions in the replay ready to issue
+        void replayReady_(const InstPtr &);
+
+        // Mark instruction as not ready and schedule replay ready
+        void updateInstReplayReady(const InstPtr &);
+
         ////////////////////////////////////////////////////////////////////////////////
         // Regular Function/Subroutine Call
         ////////////////////////////////////////////////////////////////////////////////
 
+        // Check if instruction is present in the any of the queues
+        bool instPresentInQueues(const InstPtr &inst_ptr);
+
+        // Check if the instruction is present in a specific queue
+        bool instInQueue(LoadStoreIssueQueue &queue,const InstPtr &inst_ptr);
+
+        LoadStoreInstInfoPtr createLoadStoreInst(const InstPtr &inst_ptr);
+        void allocateInstToQueues(const InstPtr &inst_ptr);
+
+        bool allOlderStoresIssued(const InstPtr &inst_ptr);
+
+        bool olderStoresIssued(LoadStoreIssueQueue &queue, const InstPtr &inst_ptr);
+
+        void readyDependentLoads(const LoadStoreInstInfoPtr &);
+
+        void abortYoungerLoads(const MemoryAccessInfoPtr &);
+
+        void deallocateYoungerLoadFromQueue(LoadStoreIssueQueue &, const InstPtr &);
+
+        // Remove instrunction from pipeline which share the same address
+        void invalidatePipeline(const InstPtr &);
```

Comment:
All private methods should be  suffixed with underscore, following the convention of the rest of the code and to enhance the readability of the code